### PR TITLE
Upgrade PyJWT  1.5.3 ->  2.0.1

### PR DESCRIPTION
**Pull request by RaduW**
This PR upgrades PyJWT to the latest version of the library.

The upgrade is necessary for using Apple App Store Connect (couldn't get to successfully login with 1.5.3).
There have been some changes in the pyJWT API from version 1.5.3 that need careful examination.

### CODE REVIEW

**Comment by mitsuhiko** [comment]
There appear to he some breakages as a result of this. For instance in `integrations/bitbucket/client.py` we assume that the JWT thing is in bytes, but it now needs to be dealt with as unicode string.

**Review by markstory — COMMENTED** [review]
The code changes look good, but you would want to verify that existing Jira webhook JWTs still work with the new code. Tests generate new JWTs and wouldn't expose compatibility issues with existing stored data.

**Comment by RaduW** [comment]
> The code changes look good, but you would want to verify that existing Jira webhook JWTs still work with the new code. Tests generate new JWTs and wouldn't expose compatibility issues with existing stored data.

@markstory good point. I was afraid of the same thing. 
Fortunately only the code decoding of the JWToken needed to be changed so the generation of JWTs hasn't changed.
Of course the lib has changed so there in not guarantee that the JWToken is the same.
I have manually checked the generation with the two lib versions and they do output identical tokens (except that in version 1.5.3 the token is a byte array and in 2.0.1 is a UTF-8 string). Since the strings only contain ASCII the two tokens are  identical.

**Review by markstory — APPROVED** [review]

**Comment by ceorourke** [comment]
I'm still working on testing Jira Server and the GitHub plugin, but I had issues when trying to use Bitbucket with this change. I went to open the "Link Bitbucket Issue" modal and it tries to make an API call to BB to get the repositories, but fails with the ApiError "Cannot decode JWT: Invalid header padding". 

Instructions for setup are here: https://develop.sentry.dev/integrations/bitbucket/

Everything else was fine, I'll report back when I've completed the tests for the remaining 2.

**Comment by flub** [comment]
Thanks for the pointer to the instructions!  Turns out one call was missed during the upgrade but the latest commit should fix this.  I also grepped for all other occurrences and it doesn't look like they have the same issue so I think that should be it.

File: src/sentry/integrations/bitbucket/client.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def request(self, method, path, data=None, params=None, **kwargs):
             "sub": self.subject,
         }
         encoded_jwt = jwt.encode(jwt_payload, self.shared_secret)
-        headers = {"Authorization": b"JWT %s" % encoded_jwt}
+        headers = {"Authorization": "JWT %s" % encoded_jwt}
```
Comment (ceorourke) [inline src/sentry/integrations/bitbucket/client.py:61]: I'm not sure this change is needed - I'm still running into the issue I mentioned before, but if I put this back to bytes it works.

Comment (flub) [reply src/sentry/integrations/bitbucket/client.py:61]: this sounds like you forgot to re-install the dependencies?  On the new version the `encoded_jwt` is a string and hence this was changed.  I don't think there's much to gain by supporting both versions at the same time so would be prefer to leave it like this.

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**